paperqa/agents/env.py

```diff
@@ -174,19 +174,23 @@ async def step(
     ) -> tuple[list[Message], float, bool, bool]:
         self.state.record_action(action)
 
-        # If the action has empty tool_calls, the agent can later take that into account
-        msgs = cast(
+        response_messages = cast(
             list[Message],
-            await self.exec_tool_calls(action, state=self.state, handle_tool_exc=True),
+            await self.exec_tool_calls(
+                action,
+                state=self.state,
+                handle_tool_exc=True,
+                ordered=True,  # prevent concurrent gather evidence steps or anything
```

Comment:
Can you do `ordered=True` in another PR?

Alternately, can you put a `TODO` here to remove this?

Reply:
Do you think it's not a bug to allow concurrent gather evidence?

Reply:
removed